Added support for google`s redirection url

pionl opened the pull request:
If google link is present, it will use the redirection uri instead the
original uri

radiovisual commented:
Awesome! Thanks for all the work. 👍 

Can you also update the README Patterns so that users can see that what this PR does?

Thanks!

pionl commented:
@radiovisual Great to hear, thank you for the response. I've updated readme, is it ok?

radiovisual commented:
Also, can you explain your usecase for these redirection urls? and why you are wanting to extract the video id from a redirection url? **I am just curious.** It's an interesting idea, so I am just curious to hear how you came to need this feature. 😄

radiovisual commented on test.js:
```diff
@@ -226,6 +226,19 @@ test('handles youtube attribution_links', t => {
 	t.is(fn('http://www.youtube.com/attribution_link?u=/watch?v=ABC12302&feature=share&list=UUsnCjinFcybOuyJU1NFOJmg&a=LjnCygXKl21WkJdyKu9O-w').service, 'youtube');
 });
 
+test('handles youtube google link', t => {
```
Can we change these to use the word "redirection" so it's more clear to future contributors or those reading the tests?

So: `'handles youtube google link'` => `handles google redirection to youtube`

radiovisual commented on test.js:
```diff
@@ -226,6 +226,19 @@ test('handles youtube attribution_links', t => {
+	t.is(fn('https://www.google.cz/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&ved=0ahUKEwj30L2MvpDVAhUFZVAKHb8CBaYQuAIIKDAB&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DtcaUcL8MiKk&usg=AFQjCNFrjHwQiaZ6q-w83GkDd9bNyRQvMw').id, 'tcaUcL8MiKk');
+});
+
+test('handles vimeo google link', t => {
```
`'handles vimeo google link'`=> `'handles google redirection to vimeo'`

pionl replied:
Of course, much better.

radiovisual commented on readme.md:
````diff
@@ -167,6 +173,12 @@ https://videopress.com/embed/*
+**Google redirection url**
+
+```
+https://google.*/url?source=web&cd=1&ved=0ahUKEwj30L2MvpDVAhUFZVAKHb8CBaYQuAIIIjAA&url=*&usg=AFQjCNECyDn3DQL7U6VW2CnXQQjB0gNKqA
````
It is misleading to use the * in the domain, because I have told the readers that the `*` means something specific,

so please change: `https://google.*` to `https://google.<extension>` or just any valid example like `https://google.cz`.

Also your example url uses the `url` parameter twice and it's really long, so I think we can condense this down to match the format of the other examples better:

```
https://google.<extension>/?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DeG1uDU0rSLw
```

pionl replied:
Thank you!

pionl commented:
I've a tool that copies a link from Google search result page (like searching for youtube video) and then I want to get the video id from the link. I also support grabbing standard youtube/vimeo link.

Of course I could detect the URL before using `getVideoId`. 

Later I will update the requested changes.

Thank you,

pionl commented:
Hi @radiovisual, thank you for the feedback. I've updated all as provided. Also I've added support for YouTube image links like `https://i.ytimg.com/vi/0okagl9U2eo/hqdefault.jpg?sqp=-oaymwEWCKgBEF5IWvKriqkDCQgBFQAAiEIYAQ==&rs=AOn4CLBQ39O8xyR-rV03CLcP92TnSjHKpQ`.

Best regards

radiovisual commented:
Awesome work. Thanks! Merging! 👍

radiovisual commented:
This functionality is now available in version `2.1.6`. 🎉
